docs(aio): add Material Community Components

tiaguinho opened the pull request:
@StephenFluin can you check this?

sliekens committed:
fix(router): cache route handle if found (#22475)

When asking the route reuse strategy to retrieve a detached route handle, store the
return value in a local variable for further processing instead of asking again later.

resolves #22474

PR Close #22475

tiaguinho committed:
docs(aio): add Material Community Components

voithos committed:
fix(router): avoid freezing queryParams in-place (#22663)

The recognizer code used to call Object.freeze() on queryParams before
using them to construct ActivatedRoutes, with the intent being to help
avoid common invalid usage. Unfortunately, Object.freeze() works
in-place, so this was also freezing the queryParams on the actual
UrlTree object, making it more difficult to manipulate UrlTrees in
things like UrlHandlingStrategy.

This change simply shallow-copies the queryParams before freezing them.

Fixes #22617

PR Close #22663

StephenFluin reviewed (approved):
From a content perspective this looks good. Not sure why CircleCI is failing though, so we need to figure that out.

kara commented:
@StephenFluin The failure is unrelated to this PR. It was a rebase error that is now fixed in master

tiaguinho commented:
Thanks, @kara.
Is everything alright then?

textbook committed:
docs(aio): update HTTP error test example (#22844)

Update the example to match the description preceding it, which refers to the
use of the error method and ErrorEvent rather than the flush method with a
non-2xx status as shown previously.

PR Close #22844

StephenFluin committed:
docs(aio): Remove Intertech with no courses scheduled (#22867)
PR Close #22867

cexbrayat committed:
fix(aio): fix error in import after RxJS 6 migration (#22886)

PR Close #22886

oburakevych committed:
docs(aio): add Angular Conf Australia to events (#22929)

Angular Conf Australia 2018 will be held at June 22 in Melbourne, Australia! 

https://www.angularconf.com.au/
PR Close #22929

davidstanke committed:
docs(bazel): add a link to the Bazel doc (#22940)

The developer doc mentions but doesn't link to BAZEL.md. Add link and fix capitalization.
PR Close #22940

jbogarthyde committed:
docs: update glossary architectural terms (#23045)

PR Close #23045

camilogiraldo committed:
docs: add missing link to bootstrapping section (#23214)

PR Close #23214

jainanuj0812 committed:
docs(forms): Fixed a typo in the reactive form (From 'address' to 'secretLairs') section (#23221)

PR Close #23221

alxhub committed:
feat(ivy): first steps towards ngtsc mode (#23455)

This commit adds a new compiler pipeline that isn't dependent on global
analysis, referred to as 'ngtsc'. This new compiler is accessed by
running ngc with "enableIvy" set to "ngtsc". It reuses the same initialization
logic but creates a new implementation of Program which does not perform the
global-level analysis that AngularCompilerProgram does. It will be the
foundation for the production Ivy compiler.

PR Close #23455

marclaval committed:
feat(ivy): support lifecycle hooks of ViewContainerRef (#23396)

PR Close #23396

mhevery committed:
perf(ivy): add performance counters in ngDevMode (#23385)

PR Close #23385

alexeagle committed:
build: serve ivy todo app with real http-server (#23446)

PR Close #23446

swseverance committed:
style: remove empty comments (#23404)

PR Close #23404

brandonroberts committed:
ci: add Brandon Roberts as an aio approver (#23417)

PR Close #23417

pkozlowski-opensource committed:
fix(ivy): properly destroy view trees where root is an embedded view without children (#23482)

The bug fixed here steams from the fact that we are traversing too far up
in the views tree hierarchy in the destroyViewTree function.

The logic in destroyViewTree is off if we start removal at an embedded view
without any child views. For such a case we should just clean up (cleanUpView)
this one view without paying attention to next / parent views.

PR Close #23482

treylitefm committed:
fix(benchpress): Fix promise chain in chrome_driver_extension. (#23458)

Occasionally the promise to clear the chrome buffer resolves after the subsequent call to start the
timer. This problem causes flakiness in our tests that rely on benchpress, usually manifesting
itself as a "Tried too often to get the ending mark: 21" error thrown by this line:

https://github.com/angular/angular/blob/master/packages/benchpress/src/metric/perflog_metric.ts#L162

PR Close #23458

mprobst committed:
fix(compiler): avoid a crash in ngc-wrapped. (#23468)

`ng.performCompilation` can return an `undefined` program, which is not handled by ngc-wrapped.

Avoid crashing by checking for the error return and returning the diagnostics.
PR Close #23468

samjulien committed:
docs(aio): Add UpgradingAngularJS to education resources (#23169)

PR Close #23169

gkalpak committed:
ci(aio): fix `deploy-to-firebase` script (#23470)

Temporary workaround for angular/angular-cli#10398.
The behavior of `yarn build` remains the same, but building for a
specific deployment env (e.g. archive, next) requires
`yarn build-for $deployEnv`.

PR Close #23470

PaulTFreedman committed:
docs(benchpress): fix typo in README (#23471) (#23488)

PR Close #23488

ocombe committed:
build(common): mark locales files as side-effect-full (#23509)

Fixes https://github.com/angular/angular-cli/issues/10322
PR Close #23509

matsko committed:
refactor(core): tree-shake application_module providers (#23477)

PR Close #23477

alexeagle committed:
ci: publish build snapshots from Bazel/CircleCI (#23512)

This uses a new script and CircleCI job called "build-packages-dist"
which shims the new Bazel build to produce outputs matching the legacy
build. We'll use this to get AIO testing onto CircleCI as well.

We move the integration tests to a new circleCI job that depends on this
one, as well as the build publishing job.

Note that every PR will have a trivial green publishing status, because
we always create this job even for PRs. We'd rather not - see
https://discuss.circleci.com/t/workflows-pull-request-filter/14396/4

PR Close #23512

sreevani-ship-it committed:
docs: fix typo (#23514)

PR Close #23514

alexeagle committed:
ci: fix snapshot publishing (#23516)

PR Close #23516

alexeagle committed:
ci: require green integration tests to publish snapshot (#23517)

Now looks like https://circleci.com/workflow-run/921ffc90-cff6-4f48-97df-740d60d5bf2b

PR Close #23517

kara committed:
feat(ivy): support injection flags at runtime (#23518)

PR Close #23518

ocombe committed:
test: add i18n to cli-hello-world integration test (#23527)

PR Close #23527

alxhub committed:
refactor(compiler): compile{Component,Directive} take only local information (#23545)

Previously, the compileComponent() and compileDirective() APIs still required
the output of global analysis, even though they only read local information
from that output.

With this refactor, compileComponent() and compileDirective() now define
their inputs explicitly, with the new interfaces R3ComponentMetadata and
R3DirectiveMetadata. compileComponentGlobal() and compileDirectiveGlobal()
are introduced and convert from global analysis output into the new metadata
format.

This refactor also splits out the view compiler into separate files as
r3_view_compiler_local.ts was getting unwieldy.

Finally, this refactor also splits out generation of DI factory functions
into a separate r3_factory utility as the logic is utilized between different
compilers.

PR Close #23545

noahmedling committed:
docs: corrected spelling of "ambient".

vicb committed:
test(ivy): add html to ivy ast transformer tests (#23546)

PR Close #23546

matsko committed:
fix(animations): retain state styling for nodes that are moved around (#23534)

PR Close #23534

vicb committed:
style: format code

gkalpak committed:
refactor(aio): include print styles last to overwrite other styles (#23538)

PR Close #23538

gkalpak committed:
refactor(aio): use the same selectors for screen and print styles (#23538)

PR Close #23538

gkalpak committed:
fix(aio): fix `code-example` print styles when printing backgrounds (#23538)

Fixes #23431

PR Close #23538

gkalpak committed:
fix(aio): remove main background color when printing (#23538)

PR Close #23538

alxhub committed:
fix(compiler-cli): don't rely on incompatible TS method (#23550)

g3 and the Angular repo have different versions of TypeScript, and
ts.updateIdentifier() has a different signature in the different versions.
There is no way to write a call to the function that will compile in both
versions simultaneously.

Instead, use ts.getMutableClone() as that has the same effect of cloning
the identifier.

PR Close #23550

jenniferfell committed:
docs: add new info about angular update policies and resources (#23551)

PR Close #23551

vikerman committed:
fix(core): avoid eager providers re-initialization (#23559)

Fix a corner case where eager providers were getting constructed twice if the provider was requested before the initialization of the NgModule is complete.

PR Close #23559

vicb committed:
docs(ivy): upddate the status (#23562)

PR Close #23562

kara committed:
fix(ivy): only generate TViews once per embedded template (#23385)

PR Close #23385

IgorMinar committed:
Revert "refactor(core): tree-shake application_module providers (#23477)"

This reverts commit eb031c6ff1d416d90d7c89dbeb367f9dc1927564.

The change is breaking targets in g3 see cl/194336387.

jbogarthyde committed:
docs: correct node.js version and usage (#23565)

PR Close #23565

jbogarthyde committed:
docs: correct typos (#23565)

PR Close #23565

jbogarthyde committed:
docs: correct more typos (#23565)

PR Close #23565

Commit:
docs(aio): update docs error in guide/http (#23567)

Updates documentation to include examples for both req.flush and
req.error in http testing examples.

PR Close #23567

gkalpak committed:
build(aio): add support for faster, unoptimized `serve` (#23569)

When running `yarn start` and `yarn serve-and-sync`, we are usually
more interested in faster re-build times than optimized builds. This was
also the behavior, before upgrading to @angular/cli@6 (fc5af69fb).

This commit introduces a new configuration (`fast`), which is used by
`yarn start` and `yarn serve-and-sync` to restore the faster,
unoptimized builds.
Other commands, such as `ng serve` and `ng e2e`, remain unchanged (using
slower, optimized builds).

PR Close #23569

IgorMinar committed:
release: cut the v6.0.0-rc.6 release

vicb committed:
feat(ivy): add error reporting to the html to ivy transformer (#23546)

PR Close #23546

gkalpak committed:
fix(service-worker): deprecate `versionedFiles` in asset-group resources (#23584)

Since `versionedFiles` behaves in the exact same way as `files`, there
is no reaason to have both. Users should use `files` instead.

This commit deprecates the property and prints a warning when coming
across an asset-group that uses it. It should be completely removed in
a future version.

Note, it has also been removed from the default `ngsw-config.json`
template in angular/devkit#754.

PR Close #23584

gkalpak committed:
refactor(service-worker): sort manifest `url`/`hashTable` entries (#23586)

This makes it easier to quickly check whether a specific file ended up
in the manifest, for example when debugging.

PR Close #23586

alexeagle committed:
ci: hide encryption key from circleci logs (#23585)

PR Close #23585

sjtrimble committed:
feat(aio): add brand and concept icons, img style class more flexible (#23589)

PR Close #23589

sarunint committed:
docs(aio): update Egghead.io URL (#23598)

Closes #23597
PR Close #23598

gkalpak committed:
docs: fix typo in tag name (`my-child` --> `app-child`) (#23606)

Fixes #23599

PR Close #23606

brandonroberts committed:
build(aio): align stackblitz files with Angular CLI V6 (#23521)

Also cleans up legacy references to `.angular-cli.json`

PR Close #23521

sreevani-ship-it committed:
docs: add doc to include updates to the index.html with the new ng add command (#23616)

PR Close #23616

gregmagolan committed:
build(bazel): update to rules_typescript 0.12.3 (#23617)

PR Close #23617

alexeagle committed:
build: update bazel to 0.13 (#23623)

PR Close #23623

Commit:
style(aio): add space between `.home` and `.hamburger` (#23624)

When the `.hamburger` icon is clicked, it's background is drawn until
the very edge of `.home`'s image, leaving no space.

PR Close #23624

gkalpak committed:
test(service-worker): improve adding clients in `SwTestHarness` (#23625)

This commits changes how clients are added in `SwTestHarness`, so that
the behavior in tests closer mimics what would happen in an actual
ServiceWorker.
It also removes auto-adding clients when calling `clients.get()`, which
could hide bugs related to non-existing clients.

PR Close #23625

gkalpak committed:
test(service-worker): support mock requests with null/empty client ID (#23625)

PR Close #23625

gkalpak committed:
fix(service-worker): correctly handle requests with empty `clientId` (#23625)

Requests from clients that are not assigned a client ID by the browser
will produce `fetch` events with `null` or empty (`''`) `clientId`s.

Previously, the ServiceWorker only handled `null` values correctly. Yet
empty strings are also valid (see for example [here][1] and [there][2]).
With this commit, the SW will interpret _all_ falsy `clientId` values
the same (i.e. "no client ID assigned") and handle them appropriately.

Related Chromium issue/discussion: [#832105][3]

[1]: https://github.com/w3c/ServiceWorker/blob/4cc72bd0f13359f16cc733d88bed4ea52ee063c0/docs/index.bs#L1392
[2]: https://w3c.github.io/ServiceWorker/#fetchevent-interface
[3]: https://bugs.chromium.org/p/chromium/issues/detail?id=832105

Fixes #23526

PR Close #23625

matsko committed:
fix(animations): properly clean up queried element styles in safari/edge (#23633)

Prior to this patch, if an element is queried and animated for 0 seconds
(just a style() call and nothing else) then the styles applied would not
be properly cleaned up due to their camelCased nature.

PR Close #23633

sreevani-ship-it committed:
docs: add information on when not to use tree-shakable providers (#23634)

PR Close #23634

gkalpak committed:
fix(aio): correctly route embedded live-example URLs from SW (#23637)

Partially addresses #23626.

PR Close #23637

andrewseguin committed:
docs(elements): add intro connecting angular elements to custom elements (#23638)

PR Close #23638

brandonroberts committed:
docs(aio): Upgrade server-side rendering example to Angular V6 (#23649)

PR Close #23649

Commit:
refactor(aio): move right margin from `.home` image to `.home` anchor (#23624)

This makes the outline of `.home` symmetric.

PR Close #23624

IgorMinar committed:
docs: update PUBLIC_API.md with the latest list of packages and clarifications

alexeagle committed:
ci: fix github_token following README (#23658)

PR Close #23658

brandonroberts committed:
docs(aio): Update i18n example to Angular V6 (#23660)

PR Close #23660

Commit:
ci: add config for size plugin of the github rebot (#23665)

PR Close #23665

petebacondarwin committed:
docs(animations): fix content errors (#23668)

PR Close #23668

petebacondarwin committed:
style(animations): fix short param names (#23668)

PR Close #23668

IgorMinar committed:
build: update to rxjs@6.0.0 (#23679)

PR Close #23679

alexeagle committed:
build: update to latest nodejs bazel rules (#23683)

PR Close #23683

jasonaden committed:
fix(router): correct the segment parsing so it won't break on ampersand (#23684)

PR Close #23684

IgorMinar committed:
release: cut the v6.0.0 release

filipesilva committed:
build(aio): update to Angular CLI 6.0.0 (#23687)

PR Close #23687

filipesilva committed:
build(aio): use Angular 6.0.0 (#23687)

PR Close #23687

IgorMinar committed:
docs: add link to the v6 release announcement to our changelog

IgorMinar committed:
docs: improve the GitHub README.md, update links, etc

IgorMinar committed:
fix(aio): correct project id for deployment of archive sites

IgorMinar committed:
build: update the scripts/release/post-check script for 6.0.x

IgorMinar committed:
feat(aio): add v6 release notification (#23690)

PR Close #23690

brandonroberts committed:
docs(aio): Upgrade example dependencies to Angular V6 (#23660)

PR Close #23660

petebacondarwin committed:
build(aio): include `navigation.json` changes in docs-watch (#23698)

Closes #23582

PR Close #23698

tiaguinho commented:
@StephenFluin, do you think I need to do some change on this PR?

IgorMinar committed:
test: fix firebase deployment script test

When I fixed the project id in 2c4850dc582287b7c34d4d26066fe4993638cbf0,
I didn't realize we had a test that verified the wrong behavior.

IgorMinar committed:
Revert "docs(animations): fix content errors (#23668)"

This reverts commit 005dc8f68b9d09f2a4db46bdbfae3e05d3604818.

The PR accidently introduced a breaking change
https://github.com/angular/angular/pull/23668#discussion_r186265055

IgorMinar committed:
Revert "style(animations): fix short param names (#23668)"

This reverts commit e3518967ad705b133ee21eedcf9546b5cb2e817e.

This PR accidentaly introduces a breaking change:
https://github.com/angular/angular/pull/23668#discussion_r186265055

alxhub committed:
fix(core): call ngOnDestroy on all services that have it (#23755)

Previously, ngOnDestroy was only called on services which were statically
determined to have ngOnDestroy methods. In some cases, such as with services
instantiated via factory functions, it's not statically known that the service
has an ngOnDestroy method.

This commit changes the runtime to look for ngOnDestroy when instantiating
all DI tokens, and to call the method if it's present.

Fixes #22466
Fixes #22240
Fixes #14818

PR Close #23755

mhartington committed:
docs(aio): add mhartington to gde (#23777)

PR Close #23777

LucasSloan committed:
build: upgrade to TypeScript 2.8 (#23782)

PR Close #23782

alcfeoh committed:
docs(aio): add Alain Chautard in GDE list (#23783)

PR Close #23783

gkalpak committed:
fix(aio): add link to v5 docs (#23794)

Fixes #23781

PR Close #23794

alexeagle committed:
ci: Remove Chuck from pullapprove (#23798)

Jason takes over his role on core, Keen for everything else

PR Close #23798

StephenFluin commented:
It looks like you added another commit. We'll need to squash / remove that before we can merge.

tiaguinho commented:
Hi @StephenFluins sorry for that. I think it's correct now.

StephenFluin reviewed (changes requested):
One tiny problem still, can you add back the newline at the end of the file so there isn't the extra change?

jbogarthyde committed:
docs(elements): emphasize future direction, update link (#23806)

PR Close #23806

jbogarthyde committed:
docs(elements): add angular element term to glossary (#23807)

PR Close #23807

jenniferfell committed:
docs: change release_schedule.md to link to new angular release page in docs (#23808)

PR Close #23808

tiaguinho commented:
Just add the line. Can I do anything more to help?

matsko committed:
feat(ivy): support injectable sanitization service (#23809)

PR Close #23809

Commit:
docs: update version to 6 in language-service (#20795)

PR Close #20795

brandonroberts committed:
docs(aio): Expose server and CLI configuration for universal in guide (#23842)

Closes #23795

PR Close #23842

gkalpak commented:
@tiaguinho, you just need to squash the commits and we're good to go :grin:

alexeagle committed:
build: update to latest TypeScript rules (#23828)

Fixes #23810

PR Close #23828

andrewseguin committed:
fix(elements): always check to create strategy (#23825)

PR Close #23825

collink committed:
docs(aio): changed 'onVoted' output property to 'voted' to be in line with the styleguide (#23832)

PR Close #23832

matsko committed:
fix(animations): do not throw errors when a destroyed component is animated (#23836)

PR Close #23836

vicb committed:
fix(compiler): generate constant array for i18n attributes (#23837)

PR Close #23837

coryrylan committed:
docs(aio): add Cory Rylan to GDE resources (#23840)

PR Close #23840

vikerman committed:
fix(elements): prevent closure renaming of platform properties (#23843)

Closure compiler with type based optimizations has a bug where externs for inherited static fields are not being honored. For Angular Elements this meant that 'observedAttributes' static field which is marked as an extern for the base HTMLElement class was getting renamed.

This commit works around the bug by using quoted access of 'observedAttributes' that explicitly prevents the renaming.

PR Close #23843

IgorMinar committed:
release: cut the v6.0.1 release

IgorMinar committed:
build: replace the old publish script with a new bazel-based one

tiaguinho commented:
@gkalpak I think we good to go now... 😃

petebacondarwin committed:
build(aio): display types of API const docs correctly (#23850)

Previously these docs always displayed `any` as the type
of the const export. Now the type is computed correctly from
the declared type or initializer of the constant.

PR Close #23850

gkalpak committed:
fix(aio): make background transparent in 144x144 PWA icon (#23851)

Fixes #23827

PR Close #23851

gkalpak commented on aio/content/marketing/resources.json:
```diff
@@ -704,3 +711,4 @@
     }
   }
 }
+
```
Where did this come from? can you remove it please :grin:

lacolaco committed:
docs(aio): fix path to observables guide (#23858)
PR Close #23858

alexeagle committed:
test: switch to ts_web_test_suite (#23859)

Unit tests now run on Firefox too

PR Close #23859

Commit:
ci: ensure github-robot listens on circleci builds (#23863)
PR Close #23863

kara committed:
fix(ivy): move next property to TNode (#23869)

PR Close #23869

petebacondarwin committed:
build(aio): improve `enum` API rendering (#23872)

* The member details section is now called "Members", rather
than "Properties".
* The property table now displays appropriate table headings:
"Member", "Value", "Description".
* The "Value" column is not shown if none of the members have
a value.

Closes #22678

PR Close #23872

JoostK committed:
perf(ivy): avoid creating bound function in pipeBind3 (#23882)

PR Close #23882

jongear committed:
docs(aio): use heroesUrl (#23884)

PR Close #23884

petebacondarwin committed:
build(aio): ensure usageNotes are copied into decorator API docs (#23901)

PR Close #23901

gregmagolan committed:
build: only match version tags for BUILD_SCM_VERSION (#23903)

PR Close #23903

jelbourn committed:
feat(platform-browser): allow lazy-loading HammerJS (#23906)

PR Close #23906

alxhub committed:
release: cut the v6.0.2 release

alexeagle committed:
test: don't run unit tests on Firefox (#23942)

PR Close #23942

brandonroberts committed:
docs(aio): Add missing dependencies and files to testing zip file download (#23948)

Closes #23060

PR Close #23948

gkalpak committed:
style(aio): remove background from `lazy-loading` concept icon (#23950)

Fixes #23938

PR Close #23950

vthinkxie committed:
docs(aio): add ant design of angular in resources (#23953)

PR Close #23953

gkalpak committed:
refactor(aio): clean up `attribute-utils` (#23960)

PR Close #23960

gkalpak committed:
fix(aio): allow setting live-example title from content (#23960)

Previously, it was possible to set the live-example title as content in
the `<live-example>` element. This relied on our custom loader
functionality that extracted the content from the DOM element before
passing it to the Angular compiler and stored it on a property for later
retrieval.
Since we switched to custom elements (and got rid of the custom loader),
the property is no longer populated with the contents. As a result, many
live examples show the default title ("live example") instead of the one
specified as content.

This commit fixes it by projecting the content into an invisible node
for later retrieval  (similar to what we do in other components, such as
the `CodeExampleComponent`).

PR Close #23960

gkalpak committed:
fix(aio): avoid unnecessary re-calculations in live-examples (#23960)

With `plnkrs`, we used to choose a different plnkr mode (normal vs
embedded) based on the size of the screen. This affected the layout of
the plnkr page ("embedded" plnkr mode was usable on small screens, while
"normal" mode wasn't). This is not to be confused with the live-example
mode we use today to determine whether the live-example should be a link
(that open StackBlitz on a new page) or embedded into the document
(using an iframe).

Since we no longer need to change the live-example URL based on the
screen size, there is no need to listen for rezise events on Window. The
necessary properties can be computed once and certain variables are
obsolete.

PR Close #23960

tama3bb committed:
docs(aio): fix typo (#23990)

are are -> are
PR Close #23990
